core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Range.hpp

```diff
@@ -58,7 +58,8 @@ template <class Functor, class Reducer, class Policy,
           bool = std::is_arithmetic_v<typename Reducer::value_type>>
 struct OpenACCParallelReduceHelper {
   OpenACCParallelReduceHelper(Functor const&, Reducer const&, Policy const&) {
-    static_assert(std::is_void_v<Functor>, "not implemented");
+    static_assert(Kokkos::Impl::always_false<Functor>::value,
+                  "not implemented");
   }
```

Comment:
Can't you just do
```suggestion
    static_assert(!Kokkos::Impl::always_true<Functor>::value,
                  "not implemented");
  }
```
and avoid introducing more code?